WooCommerce/Classes/ViewRelated/Upgrades/UpgradesViewModel.swift

```diff
@@ -0,0 +1,187 @@
+            if daysLeft(for: plan) > 0 {
+                return Localization.freeTrial
+            } else {
+                return Localization.trialEnded
+            }
+        }
+
+        // For non-free trials plans  remove any mention to WPCom.
+        let toRemove = "WordPress.com"
+        let sanitizedName = plan.name.replacingOccurrences(of: toRemove, with: "").trimmingCharacters(in: .whitespacesAndNewlines)
+        return sanitizedName
+    }
+
+    /// Returns a plan specific details information.
+    ///
+    static func getPlanInfo(from plan: WPComSitePlan) -> String {
+        let daysLeft = daysLeft(for: plan)
+        let planDuration = planDurationInDays(for: plan)
+
+        if plan.isFreeTrial {
+            if daysLeft > 0 {
+                return Localization.freeTrialPlanInfo(planDuration: planDuration, daysLeft: daysLeft)
+            } else {
+                return Localization.trialEndedInfo
+            }
+        }
+
+        let planName = getPlanName(from: plan)
+        guard let expireDate = plan.expiryDate else {
+            return ""
+        }
+
+        let expireText = DateFormatter.mediumLengthLocalizedDateFormatter.string(from: expireDate)
+        return Localization.planInfo(planName: planName, expirationDate: expireText)
+    }
+
+    /// Only allow to upgrade the plan if we are on a free trial.
+    ///
+    static func getUpgradeNowButtonVisibility(from plan: WPComSitePlan) -> Bool {
+        plan.isFreeTrial
+    }
+
+    /// Returns a site plan duration in days.
+    ///
+    static func planDurationInDays(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        guard let subscribedDate = plan.subscribedDate?.startOfDay(timezone: .current),
+              let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let duration = Calendar.current.dateComponents([.day], from: subscribedDate, to: expiryDate).day ?? 0
+        return duration
+    }
+
+    /// Returns how many days site  plan has left.
+    ///
+    static func daysLeft(for plan: WPComSitePlan) -> Int {
+        // Normalize dates in the same timezone.
+        let today = Date().startOfDay(timezone: .current)
+        guard let expiryDate = plan.expiryDate?.startOfDay(timezone: .current) else {
+            return 0
+        }
+
+        let daysLeft = Calendar.current.dateComponents([.day], from: today, to: expiryDate).day ?? 0
+        return daysLeft
+    }
+}
+
+// MARK: Definitions
+private extension UpgradesViewModel {
+    enum Localization {
+        static let freeTrial = NSLocalizedString("Free Trial", comment: "Plan name for an active free trial")
+        static let trialEnded = NSLocalizedString("Trial ended", comment: "Plan name for an expired free trial")
+        static let trialEndedInfo = NSLocalizedString("Your free trial has ended and have limited access to all the features. Subscribe to eCommerce now.",
```

Comment:
There's a small grammar issue in this sentence. We need a subject for the second part of the sentence, e.g. "you" or "your store":

```suggestion
        static let trialEndedInfo = NSLocalizedString("Your free trial has ended and you have limited access to all the features. Subscribe to eCommerce now.",
```

For comparison, here's the string that's used for expired free trials in Calypso:

> Your free trial has expired. Upgrade to a plan to unlock new features and start selling.

Reply:
Thanks Rachel, I just copied the strings from figma and didn't read them into detail, shame on me 😞